File: metaphacts-platform/web/src/test/mocks/ThumbnailService.ts
/*
 * Copyright (C) 2015-2019, metaphacts GmbH
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2.1 of the License, or (at your option) any later version.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, you can receive a copy
 * of the GNU Lesser General Public License from http://www.gnu.org/
 */

import { expect } from 'chai';

export function mockThumbnailService(
  server: sinon.SinonFakeServer, response: string, expectedRequest?: string
) {
  server.respondWith(
    'POST', '/rest/data/rdf/utils/thumbnails/default?repository=default',
    (xhr: sinon.SinonFakeXMLHttpRequest) => {
      if (expectedRequest) {
        expect(xhr.requestBody).to.be.equalIgnoreSpaces(expectedRequest);
      }

      xhr.respond(
        200, { 'Content-Type': 'application/json' }, response
      );
    }
  );
}

File: src/main/java/org/researchspace/thumbnails/ThumbnailService.java
/**
 * ResearchSpace
 * Copyright (C) 2020, © Trustees of the British Museum
 * Copyright (C) 2015-2019, metaphacts GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.

 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package org.researchspace.thumbnails;

import org.eclipse.rdf4j.model.IRI;
import org.eclipse.rdf4j.model.Value;
import org.eclipse.rdf4j.repository.Repository;

import java.util.Map;
import java.util.Optional;

/**
 * @author Alexey Morozov
 */
public interface ThumbnailService {
    /**
     * @return unique per-thumbnail-service name
     */
    String getThumbnailServiceName();

    Map<IRI, Optional<Value>> getThumbnails(Repository repository, Iterable<? extends IRI> resourceIRIs);
}
